From 248280d92b95ddc8518797b095aec1ce275e4d10 Mon Sep 17 00:00:00 2001
From: Marcel van Lohuizen <mpvl@golang.org>
Date: Mon, 17 Aug 2020 19:10:12 +0200
Subject: [PATCH] unicode: upgrade to Unicode 13.0.0

# AWS EKS
Backported To: go-1.15.15-eks
Backported On: Fri, 21 Oct 2022
Backported By: rcrozean@amazon.com
Backported From: release-branch.go1.16
Source Commit:  https://github.com/golang/go/commit/8ec5a052ec8d4e628353fb5099b29341fae9b3a4
EKS Patch Source Commit: https://github.com/rcrozean/go/commit/248280d92b95ddc8518797b095aec1ce275e4d10

This fix was required for the CVE-2022-41715 fix. The change in Unicode
cause the regexp rune size calculation to be off an fail to brake as
expected.

# Original Information

Fixes #40755

Change-Id: I14b3977317994095db8ae1bd873c174641209356
Reviewed-on: https://go-review.googlesource.com/c/go/+/248765
Run-TryBot: Marcel van Lohuizen <mpvl@golang.org>
TryBot-Result: Gobot Gobot <gobot@golang.org>
Reviewed-by: Ian Lance Taylor <iant@golang.org>
---
 api/except.txt         |   1 +
 api/next.txt           |   5 +
 src/strconv/isprint.go |  98 ++++----
 src/unicode/tables.go  | 544 ++++++++++++++++++++++++++---------------
 4 files changed, 405 insertions(+), 243 deletions(-)

diff --git a/api/except.txt b/api/except.txt
index ccfdf06c55..962bb14271 100644
--- a/api/except.txt
+++ b/api/except.txt
@@ -456,3 +456,4 @@ pkg syscall (freebsd-arm-cgo), type Statfs_t struct, Mntonname [88]int8
 pkg text/scanner, const GoTokens = 1012
 pkg unicode, const Version = "10.0.0"
 pkg unicode, const Version = "11.0.0"
+pkg unicode, const Version = "12.0.0"
diff --git a/api/next.txt b/api/next.txt
index e69de29bb2..fe7509bf82 100644
--- a/api/next.txt
+++ b/api/next.txt
@@ -0,0 +1,5 @@
+pkg unicode, const Version = "13.0.0"
+pkg unicode, var Chorasmian *RangeTable
+pkg unicode, var Dives_Akuru *RangeTable
+pkg unicode, var Khitan_Small_Script *RangeTable
+pkg unicode, var Yezidi *RangeTable
diff --git a/src/strconv/isprint.go b/src/strconv/isprint.go
index 7ada2d1b04..994a8e423c 100644
--- a/src/strconv/isprint.go
+++ b/src/strconv/isprint.go
@@ -6,7 +6,7 @@
 
 package strconv
 
-// (442+132+90)*2 + (450)*4 = 3128 bytes
+// (434+132+95)*2 + (468)*4 = 3194 bytes
 
 var isPrint16 = []uint16{
 	0x0020, 0x007e,
@@ -25,7 +25,7 @@ var isPrint16 = []uint16{
 	0x07fd, 0x082d,
 	0x0830, 0x085b,
 	0x085e, 0x086a,
-	0x08a0, 0x08bd,
+	0x08a0, 0x08c7,
 	0x08d3, 0x098c,
 	0x098f, 0x0990,
 	0x0993, 0x09b2,
@@ -56,7 +56,7 @@ var isPrint16 = []uint16{
 	0x0b3c, 0x0b44,
 	0x0b47, 0x0b48,
 	0x0b4b, 0x0b4d,
-	0x0b56, 0x0b57,
+	0x0b55, 0x0b57,
 	0x0b5c, 0x0b63,
 	0x0b66, 0x0b77,
 	0x0b82, 0x0b8a,
@@ -82,8 +82,7 @@ var isPrint16 = []uint16{
 	0x0ce6, 0x0cf2,
 	0x0d00, 0x0d4f,
 	0x0d54, 0x0d63,
-	0x0d66, 0x0d7f,
-	0x0d82, 0x0d96,
+	0x0d66, 0x0d96,
 	0x0d9a, 0x0dbd,
 	0x0dc0, 0x0dc6,
 	0x0dca, 0x0dca,
@@ -138,7 +137,7 @@ var isPrint16 = []uint16{
 	0x1a7f, 0x1a89,
 	0x1a90, 0x1a99,
 	0x1aa0, 0x1aad,
-	0x1ab0, 0x1abe,
+	0x1ab0, 0x1ac0,
 	0x1b00, 0x1b4b,
 	0x1b50, 0x1b7c,
 	0x1b80, 0x1bf3,
@@ -166,30 +165,27 @@ var isPrint16 = []uint16{
 	0x2190, 0x2426,
 	0x2440, 0x244a,
 	0x2460, 0x2b73,
-	0x2b76, 0x2b95,
-	0x2b98, 0x2cf3,
+	0x2b76, 0x2cf3,
 	0x2cf9, 0x2d27,
 	0x2d2d, 0x2d2d,
 	0x2d30, 0x2d67,
 	0x2d6f, 0x2d70,
 	0x2d7f, 0x2d96,
-	0x2da0, 0x2e4f,
+	0x2da0, 0x2e52,
 	0x2e80, 0x2ef3,
 	0x2f00, 0x2fd5,
 	0x2ff0, 0x2ffb,
 	0x3001, 0x3096,
 	0x3099, 0x30ff,
-	0x3105, 0x31ba,
-	0x31c0, 0x31e3,
-	0x31f0, 0x4db5,
-	0x4dc0, 0x9fef,
+	0x3105, 0x31e3,
+	0x31f0, 0x9ffc,
 	0xa000, 0xa48c,
 	0xa490, 0xa4c6,
 	0xa4d0, 0xa62b,
 	0xa640, 0xa6f7,
 	0xa700, 0xa7bf,
-	0xa7c2, 0xa7c6,
-	0xa7f7, 0xa82b,
+	0xa7c2, 0xa7ca,
+	0xa7f5, 0xa82c,
 	0xa830, 0xa839,
 	0xa840, 0xa877,
 	0xa880, 0xa8c5,
@@ -205,7 +201,7 @@ var isPrint16 = []uint16{
 	0xab01, 0xab06,
 	0xab09, 0xab0e,
 	0xab11, 0xab16,
-	0xab20, 0xab67,
+	0xab20, 0xab6b,
 	0xab70, 0xabed,
 	0xabf0, 0xabf9,
 	0xac00, 0xd7a3,
@@ -288,11 +284,11 @@ var isNotPrint16 = []uint16{
 	0x0cc9,
 	0x0cdf,
 	0x0cf0,
-	0x0d04,
 	0x0d0d,
 	0x0d11,
 	0x0d45,
 	0x0d49,
+	0x0d80,
 	0x0d84,
 	0x0db2,
 	0x0dbc,
@@ -335,6 +331,7 @@ var isNotPrint16 = []uint16{
 	0x1fdc,
 	0x1ff5,
 	0x208f,
+	0x2b96,
 	0x2c2f,
 	0x2c5f,
 	0x2d26,
@@ -351,7 +348,6 @@ var isNotPrint16 = []uint16{
 	0x3130,
 	0x318f,
 	0x321f,
-	0x32ff,
 	0xa9ce,
 	0xa9ff,
 	0xab27,
@@ -373,7 +369,7 @@ var isPrint32 = []uint32{
 	0x010080, 0x0100fa,
 	0x010100, 0x010102,
 	0x010107, 0x010133,
-	0x010137, 0x01019b,
+	0x010137, 0x01019c,
 	0x0101a0, 0x0101a0,
 	0x0101d0, 0x0101fd,
 	0x010280, 0x01029c,
@@ -424,19 +420,20 @@ var isPrint32 = []uint32{
 	0x010cc0, 0x010cf2,
 	0x010cfa, 0x010d27,
 	0x010d30, 0x010d39,
-	0x010e60, 0x010e7e,
+	0x010e60, 0x010ead,
+	0x010eb0, 0x010eb1,
 	0x010f00, 0x010f27,
 	0x010f30, 0x010f59,
+	0x010fb0, 0x010fcb,
 	0x010fe0, 0x010ff6,
 	0x011000, 0x01104d,
 	0x011052, 0x01106f,
 	0x01107f, 0x0110c1,
 	0x0110d0, 0x0110e8,
 	0x0110f0, 0x0110f9,
-	0x011100, 0x011146,
+	0x011100, 0x011147,
 	0x011150, 0x011176,
-	0x011180, 0x0111cd,
-	0x0111d0, 0x0111f4,
+	0x011180, 0x0111f4,
 	0x011200, 0x01123e,
 	0x011280, 0x0112a9,
 	0x0112b0, 0x0112ea,
@@ -451,7 +448,7 @@ var isPrint32 = []uint32{
 	0x01135d, 0x011363,
 	0x011366, 0x01136c,
 	0x011370, 0x011374,
-	0x011400, 0x01145f,
+	0x011400, 0x011461,
 	0x011480, 0x0114c7,
 	0x0114d0, 0x0114d9,
 	0x011580, 0x0115b5,
@@ -466,7 +463,11 @@ var isPrint32 = []uint32{
 	0x011730, 0x01173f,
 	0x011800, 0x01183b,
 	0x0118a0, 0x0118f2,
-	0x0118ff, 0x0118ff,
+	0x0118ff, 0x011906,
+	0x011909, 0x011909,
+	0x01190c, 0x011938,
+	0x01193b, 0x011946,
+	0x011950, 0x011959,
 	0x0119a0, 0x0119a7,
 	0x0119aa, 0x0119d7,
 	0x0119da, 0x0119e4,
@@ -483,6 +484,7 @@ var isPrint32 = []uint32{
 	0x011d60, 0x011d98,
 	0x011da0, 0x011da9,
 	0x011ee0, 0x011ef8,
+	0x011fb0, 0x011fb0,
 	0x011fc0, 0x011ff1,
 	0x011fff, 0x012399,
 	0x012400, 0x012474,
@@ -501,9 +503,11 @@ var isPrint32 = []uint32{
 	0x016f00, 0x016f4a,
 	0x016f4f, 0x016f87,
 	0x016f8f, 0x016f9f,
-	0x016fe0, 0x016fe3,
+	0x016fe0, 0x016fe4,
+	0x016ff0, 0x016ff1,
 	0x017000, 0x0187f7,
-	0x018800, 0x018af2,
+	0x018800, 0x018cd5,
+	0x018d00, 0x018d08,
 	0x01b000, 0x01b11e,
 	0x01b150, 0x01b152,
 	0x01b164, 0x01b167,
@@ -557,17 +561,15 @@ var isPrint32 = []uint32{
 	0x01f030, 0x01f093,
 	0x01f0a0, 0x01f0ae,
 	0x01f0b1, 0x01f0f5,
-	0x01f100, 0x01f10c,
-	0x01f110, 0x01f16c,
-	0x01f170, 0x01f1ac,
+	0x01f100, 0x01f1ad,
 	0x01f1e6, 0x01f202,
 	0x01f210, 0x01f23b,
 	0x01f240, 0x01f248,
 	0x01f250, 0x01f251,
 	0x01f260, 0x01f265,
-	0x01f300, 0x01f6d5,
+	0x01f300, 0x01f6d7,
 	0x01f6e0, 0x01f6ec,
-	0x01f6f0, 0x01f6fa,
+	0x01f6f0, 0x01f6fc,
 	0x01f700, 0x01f773,
 	0x01f780, 0x01f7d8,
 	0x01f7e0, 0x01f7eb,
@@ -576,22 +578,25 @@ var isPrint32 = []uint32{
 	0x01f850, 0x01f859,
 	0x01f860, 0x01f887,
 	0x01f890, 0x01f8ad,
-	0x01f900, 0x01f976,
-	0x01f97a, 0x01f9a2,
-	0x01f9a5, 0x01f9aa,
-	0x01f9ae, 0x01f9ca,
-	0x01f9cd, 0x01fa53,
+	0x01f8b0, 0x01f8b1,
+	0x01f900, 0x01fa53,
 	0x01fa60, 0x01fa6d,
-	0x01fa70, 0x01fa73,
+	0x01fa70, 0x01fa74,
 	0x01fa78, 0x01fa7a,
-	0x01fa80, 0x01fa82,
-	0x01fa90, 0x01fa95,
-	0x020000, 0x02a6d6,
+	0x01fa80, 0x01fa86,
+	0x01fa90, 0x01faa8,
+	0x01fab0, 0x01fab6,
+	0x01fac0, 0x01fac2,
+	0x01fad0, 0x01fad6,
+	0x01fb00, 0x01fbca,
+	0x01fbf0, 0x01fbf9,
+	0x020000, 0x02a6dd,
 	0x02a700, 0x02b734,
 	0x02b740, 0x02b81d,
 	0x02b820, 0x02cea1,
 	0x02ceb0, 0x02ebe0,
 	0x02f800, 0x02fa1d,
+	0x030000, 0x03134a,
 	0x0e0100, 0x0e01ef,
 }
 
@@ -609,6 +614,8 @@ var isNotPrint32 = []uint16{ // add 0x10000 to each entry
 	0x0a04,
 	0x0a14,
 	0x0a18,
+	0x0e7f,
+	0x0eaa,
 	0x10bd,
 	0x1135,
 	0x11e0,
@@ -622,8 +629,10 @@ var isNotPrint32 = []uint16{ // add 0x10000 to each entry
 	0x1331,
 	0x1334,
 	0x133a,
-	0x145a,
 	0x145c,
+	0x1914,
+	0x1917,
+	0x1936,
 	0x1c09,
 	0x1c37,
 	0x1ca8,
@@ -684,8 +693,9 @@ var isNotPrint32 = []uint16{ // add 0x10000 to each entry
 	0xeeaa,
 	0xf0c0,
 	0xf0d0,
-	0xf90c,
-	0xf972,
+	0xf979,
+	0xf9cc,
+	0xfb93,
 }
 
 // isGraphic lists the graphic runes not matched by IsPrint.
diff --git a/src/unicode/tables.go b/src/unicode/tables.go
index 12441682d0..a9b23bfacd 100644
--- a/src/unicode/tables.go
+++ b/src/unicode/tables.go
@@ -3,7 +3,7 @@
 package unicode
 
 // Version is the Unicode edition from which the tables are derived.
-const Version = "12.0.0"
+const Version = "13.0.0"
 
 // Categories is the set of Unicode category tables.
 var Categories = map[string]*RangeTable{
@@ -170,7 +170,7 @@ var _L = &RangeTable{
 		{0x0841, 0x0858, 1},
 		{0x0860, 0x086a, 1},
 		{0x08a0, 0x08b4, 1},
-		{0x08b6, 0x08bd, 1},
+		{0x08b6, 0x08c7, 1},
 		{0x0904, 0x0939, 1},
 		{0x093d, 0x0950, 19},
 		{0x0958, 0x0961, 1},
@@ -241,7 +241,7 @@ var _L = &RangeTable{
 		{0x0cbd, 0x0cde, 33},
 		{0x0ce0, 0x0ce1, 1},
 		{0x0cf1, 0x0cf2, 1},
-		{0x0d05, 0x0d0c, 1},
+		{0x0d04, 0x0d0c, 1},
 		{0x0d0e, 0x0d10, 1},
 		{0x0d12, 0x0d3a, 1},
 		{0x0d3d, 0x0d4e, 17},
@@ -402,10 +402,10 @@ var _L = &RangeTable{
 		{0x30fc, 0x30ff, 1},
 		{0x3105, 0x312f, 1},
 		{0x3131, 0x318e, 1},
-		{0x31a0, 0x31ba, 1},
+		{0x31a0, 0x31bf, 1},
 		{0x31f0, 0x31ff, 1},
-		{0x3400, 0x4db5, 1},
-		{0x4e00, 0x9fef, 1},
+		{0x3400, 0x4dbf, 1},
+		{0x4e00, 0x9ffc, 1},
 		{0xa000, 0xa48c, 1},
 		{0xa4d0, 0xa4fd, 1},
 		{0xa500, 0xa60c, 1},
@@ -417,8 +417,8 @@ var _L = &RangeTable{
 		{0xa717, 0xa71f, 1},
 		{0xa722, 0xa788, 1},
 		{0xa78b, 0xa7bf, 1},
-		{0xa7c2, 0xa7c6, 1},
-		{0xa7f7, 0xa801, 1},
+		{0xa7c2, 0xa7ca, 1},
+		{0xa7f5, 0xa801, 1},
 		{0xa803, 0xa805, 1},
 		{0xa807, 0xa80a, 1},
 		{0xa80c, 0xa822, 1},
@@ -454,7 +454,7 @@ var _L = &RangeTable{
 		{0xab20, 0xab26, 1},
 		{0xab28, 0xab2e, 1},
 		{0xab30, 0xab5a, 1},
-		{0xab5c, 0xab67, 1},
+		{0xab5c, 0xab69, 1},
 		{0xab70, 0xabe2, 1},
 		{0xac00, 0xd7a3, 1},
 		{0xd7b0, 0xd7c6, 1},
@@ -540,16 +540,19 @@ var _L = &RangeTable{
 		{0x10c80, 0x10cb2, 1},
 		{0x10cc0, 0x10cf2, 1},
 		{0x10d00, 0x10d23, 1},
+		{0x10e80, 0x10ea9, 1},
+		{0x10eb0, 0x10eb1, 1},
 		{0x10f00, 0x10f1c, 1},
 		{0x10f27, 0x10f30, 9},
 		{0x10f31, 0x10f45, 1},
+		{0x10fb0, 0x10fc4, 1},
 		{0x10fe0, 0x10ff6, 1},
 		{0x11003, 0x11037, 1},
 		{0x11083, 0x110af, 1},
 		{0x110d0, 0x110e8, 1},
 		{0x11103, 0x11126, 1},
-		{0x11144, 0x11150, 12},
-		{0x11151, 0x11172, 1},
+		{0x11144, 0x11147, 3},
+		{0x11150, 0x11172, 1},
 		{0x11176, 0x11183, 13},
 		{0x11184, 0x111b2, 1},
 		{0x111c1, 0x111c4, 1},
@@ -572,8 +575,8 @@ var _L = &RangeTable{
 		{0x1135d, 0x11361, 1},
 		{0x11400, 0x11434, 1},
 		{0x11447, 0x1144a, 1},
-		{0x1145f, 0x11480, 33},
-		{0x11481, 0x114af, 1},
+		{0x1145f, 0x11461, 1},
+		{0x11480, 0x114af, 1},
 		{0x114c4, 0x114c5, 1},
 		{0x114c7, 0x11580, 185},
 		{0x11581, 0x115ae, 1},
@@ -585,8 +588,13 @@ var _L = &RangeTable{
 		{0x11701, 0x1171a, 1},
 		{0x11800, 0x1182b, 1},
 		{0x118a0, 0x118df, 1},
-		{0x118ff, 0x119a0, 161},
-		{0x119a1, 0x119a7, 1},
+		{0x118ff, 0x11906, 1},
+		{0x11909, 0x1190c, 3},
+		{0x1190d, 0x11913, 1},
+		{0x11915, 0x11916, 1},
+		{0x11918, 0x1192f, 1},
+		{0x1193f, 0x11941, 2},
+		{0x119a0, 0x119a7, 1},
 		{0x119aa, 0x119d0, 1},
 		{0x119e1, 0x119e3, 2},
 		{0x11a00, 0x11a0b, 11},
@@ -608,7 +616,8 @@ var _L = &RangeTable{
 		{0x11d6a, 0x11d89, 1},
 		{0x11d98, 0x11ee0, 328},
 		{0x11ee1, 0x11ef2, 1},
-		{0x12000, 0x12399, 1},
+		{0x11fb0, 0x12000, 80},
+		{0x12001, 0x12399, 1},
 		{0x12480, 0x12543, 1},
 		{0x13000, 0x1342e, 1},
 		{0x14400, 0x14646, 1},
@@ -626,7 +635,8 @@ var _L = &RangeTable{
 		{0x16fe0, 0x16fe1, 1},
 		{0x16fe3, 0x17000, 29},
 		{0x17001, 0x187f7, 1},
-		{0x18800, 0x18af2, 1},
+		{0x18800, 0x18cd5, 1},
+		{0x18d00, 0x18d08, 1},
 		{0x1b000, 0x1b11e, 1},
 		{0x1b150, 0x1b152, 1},
 		{0x1b164, 0x1b167, 1},
@@ -696,12 +706,13 @@ var _L = &RangeTable{
 		{0x1eea1, 0x1eea3, 1},
 		{0x1eea5, 0x1eea9, 1},
 		{0x1eeab, 0x1eebb, 1},
-		{0x20000, 0x2a6d6, 1},
+		{0x20000, 0x2a6dd, 1},
 		{0x2a700, 0x2b734, 1},
 		{0x2b740, 0x2b81d, 1},
 		{0x2b820, 0x2cea1, 1},
 		{0x2ceb0, 0x2ebe0, 1},
 		{0x2f800, 0x2fa1d, 1},
+		{0x30000, 0x3134a, 1},
 	},
 	LatinOffset: 6,
 }
@@ -821,9 +832,11 @@ var _Ll = &RangeTable{
 		{0xa797, 0xa7a9, 2},
 		{0xa7af, 0xa7b5, 6},
 		{0xa7b7, 0xa7bf, 2},
-		{0xa7c3, 0xa7fa, 55},
-		{0xab30, 0xab5a, 1},
-		{0xab60, 0xab67, 1},
+		{0xa7c3, 0xa7c8, 5},
+		{0xa7ca, 0xa7f6, 44},
+		{0xa7fa, 0xab30, 822},
+		{0xab31, 0xab5a, 1},
+		{0xab60, 0xab68, 1},
 		{0xab70, 0xabbf, 1},
 		{0xfb00, 0xfb06, 1},
 		{0xfb13, 0xfb17, 1},
@@ -908,8 +921,8 @@ var _Lm = &RangeTable{
 		{0xaa70, 0xaadd, 109},
 		{0xaaf3, 0xaaf4, 1},
 		{0xab5c, 0xab5f, 1},
-		{0xff70, 0xff9e, 46},
-		{0xff9f, 0xff9f, 1},
+		{0xab69, 0xff70, 21511},
+		{0xff9e, 0xff9f, 1},
 	},
 	R32: []Range32{
 		{0x16b40, 0x16b43, 1},
@@ -945,7 +958,7 @@ var _Lo = &RangeTable{
 		{0x0840, 0x0858, 1},
 		{0x0860, 0x086a, 1},
 		{0x08a0, 0x08b4, 1},
-		{0x08b6, 0x08bd, 1},
+		{0x08b6, 0x08c7, 1},
 		{0x0904, 0x0939, 1},
 		{0x093d, 0x0950, 19},
 		{0x0958, 0x0961, 1},
@@ -1016,7 +1029,7 @@ var _Lo = &RangeTable{
 		{0x0cbd, 0x0cde, 33},
 		{0x0ce0, 0x0ce1, 1},
 		{0x0cf1, 0x0cf2, 1},
-		{0x0d05, 0x0d0c, 1},
+		{0x0d04, 0x0d0c, 1},
 		{0x0d0e, 0x0d10, 1},
 		{0x0d12, 0x0d3a, 1},
 		{0x0d3d, 0x0d4e, 17},
@@ -1127,10 +1140,10 @@ var _Lo = &RangeTable{
 		{0x30ff, 0x3105, 6},
 		{0x3106, 0x312f, 1},
 		{0x3131, 0x318e, 1},
-		{0x31a0, 0x31ba, 1},
+		{0x31a0, 0x31bf, 1},
 		{0x31f0, 0x31ff, 1},
-		{0x3400, 0x4db5, 1},
-		{0x4e00, 0x9fef, 1},
+		{0x3400, 0x4dbf, 1},
+		{0x4e00, 0x9ffc, 1},
 		{0xa000, 0xa014, 1},
 		{0xa016, 0xa48c, 1},
 		{0xa4d0, 0xa4f7, 1},
@@ -1254,16 +1267,19 @@ var _Lo = &RangeTable{
 		{0x10b80, 0x10b91, 1},
 		{0x10c00, 0x10c48, 1},
 		{0x10d00, 0x10d23, 1},
+		{0x10e80, 0x10ea9, 1},
+		{0x10eb0, 0x10eb1, 1},
 		{0x10f00, 0x10f1c, 1},
 		{0x10f27, 0x10f30, 9},
 		{0x10f31, 0x10f45, 1},
+		{0x10fb0, 0x10fc4, 1},
 		{0x10fe0, 0x10ff6, 1},
 		{0x11003, 0x11037, 1},
 		{0x11083, 0x110af, 1},
 		{0x110d0, 0x110e8, 1},
 		{0x11103, 0x11126, 1},
-		{0x11144, 0x11150, 12},
-		{0x11151, 0x11172, 1},
+		{0x11144, 0x11147, 3},
+		{0x11150, 0x11172, 1},
 		{0x11176, 0x11183, 13},
 		{0x11184, 0x111b2, 1},
 		{0x111c1, 0x111c4, 1},
@@ -1286,8 +1302,8 @@ var _Lo = &RangeTable{
 		{0x1135d, 0x11361, 1},
 		{0x11400, 0x11434, 1},
 		{0x11447, 0x1144a, 1},
-		{0x1145f, 0x11480, 33},
-		{0x11481, 0x114af, 1},
+		{0x1145f, 0x11461, 1},
+		{0x11480, 0x114af, 1},
 		{0x114c4, 0x114c5, 1},
 		{0x114c7, 0x11580, 185},
 		{0x11581, 0x115ae, 1},
@@ -1298,8 +1314,13 @@ var _Lo = &RangeTable{
 		{0x116b8, 0x11700, 72},
 		{0x11701, 0x1171a, 1},
 		{0x11800, 0x1182b, 1},
-		{0x118ff, 0x119a0, 161},
-		{0x119a1, 0x119a7, 1},
+		{0x118ff, 0x11906, 1},
+		{0x11909, 0x1190c, 3},
+		{0x1190d, 0x11913, 1},
+		{0x11915, 0x11916, 1},
+		{0x11918, 0x1192f, 1},
+		{0x1193f, 0x11941, 2},
+		{0x119a0, 0x119a7, 1},
 		{0x119aa, 0x119d0, 1},
 		{0x119e1, 0x119e3, 2},
 		{0x11a00, 0x11a0b, 11},
@@ -1321,7 +1342,8 @@ var _Lo = &RangeTable{
 		{0x11d6a, 0x11d89, 1},
 		{0x11d98, 0x11ee0, 328},
 		{0x11ee1, 0x11ef2, 1},
-		{0x12000, 0x12399, 1},
+		{0x11fb0, 0x12000, 80},
+		{0x12001, 0x12399, 1},
 		{0x12480, 0x12543, 1},
 		{0x13000, 0x1342e, 1},
 		{0x14400, 0x14646, 1},
@@ -1334,7 +1356,8 @@ var _Lo = &RangeTable{
 		{0x16f00, 0x16f4a, 1},
 		{0x16f50, 0x17000, 176},
 		{0x17001, 0x187f7, 1},
-		{0x18800, 0x18af2, 1},
+		{0x18800, 0x18cd5, 1},
+		{0x18d00, 0x18d08, 1},
 		{0x1b000, 0x1b11e, 1},
 		{0x1b150, 0x1b152, 1},
 		{0x1b164, 0x1b167, 1},
@@ -1371,12 +1394,13 @@ var _Lo = &RangeTable{
 		{0x1eea1, 0x1eea3, 1},
 		{0x1eea5, 0x1eea9, 1},
 		{0x1eeab, 0x1eebb, 1},
-		{0x20000, 0x2a6d6, 1},
+		{0x20000, 0x2a6dd, 1},
 		{0x2a700, 0x2b734, 1},
 		{0x2b740, 0x2b81d, 1},
 		{0x2b820, 0x2cea1, 1},
 		{0x2ceb0, 0x2ebe0, 1},
 		{0x2f800, 0x2fa1d, 1},
+		{0x30000, 0x3134a, 1},
 	},
 	LatinOffset: 1,
 }
@@ -1500,7 +1524,8 @@ var _Lu = &RangeTable{
 		{0xa7b0, 0xa7b4, 1},
 		{0xa7b6, 0xa7be, 2},
 		{0xa7c2, 0xa7c4, 2},
-		{0xa7c5, 0xa7c6, 1},
+		{0xa7c5, 0xa7c7, 1},
+		{0xa7c9, 0xa7f5, 44},
 		{0xff21, 0xff3a, 1},
 	},
 	R32: []Range32{
@@ -1601,7 +1626,7 @@ var _M = &RangeTable{
 		{0x0b3f, 0x0b44, 1},
 		{0x0b47, 0x0b48, 1},
 		{0x0b4b, 0x0b4d, 1},
-		{0x0b56, 0x0b57, 1},
+		{0x0b55, 0x0b57, 1},
 		{0x0b62, 0x0b63, 1},
 		{0x0b82, 0x0bbe, 60},
 		{0x0bbf, 0x0bc2, 1},
@@ -1627,9 +1652,10 @@ var _M = &RangeTable{
 		{0x0d46, 0x0d48, 1},
 		{0x0d4a, 0x0d4d, 1},
 		{0x0d57, 0x0d62, 11},
-		{0x0d63, 0x0d82, 31},
-		{0x0d83, 0x0dca, 71},
-		{0x0dcf, 0x0dd4, 1},
+		{0x0d63, 0x0d81, 30},
+		{0x0d82, 0x0d83, 1},
+		{0x0dca, 0x0dcf, 5},
+		{0x0dd0, 0x0dd4, 1},
 		{0x0dd6, 0x0dd8, 2},
 		{0x0dd9, 0x0ddf, 1},
 		{0x0df2, 0x0df3, 1},
@@ -1672,7 +1698,7 @@ var _M = &RangeTable{
 		{0x1a55, 0x1a5e, 1},
 		{0x1a60, 0x1a7c, 1},
 		{0x1a7f, 0x1ab0, 49},
-		{0x1ab1, 0x1abe, 1},
+		{0x1ab1, 0x1ac0, 1},
 		{0x1b00, 0x1b04, 1},
 		{0x1b34, 0x1b44, 1},
 		{0x1b6b, 0x1b73, 1},
@@ -1699,8 +1725,9 @@ var _M = &RangeTable{
 		{0xa802, 0xa806, 4},
 		{0xa80b, 0xa823, 24},
 		{0xa824, 0xa827, 1},
-		{0xa880, 0xa881, 1},
-		{0xa8b4, 0xa8c5, 1},
+		{0xa82c, 0xa880, 84},
+		{0xa881, 0xa8b4, 51},
+		{0xa8b5, 0xa8c5, 1},
 		{0xa8e0, 0xa8f1, 1},
 		{0xa8ff, 0xa926, 39},
 		{0xa927, 0xa92d, 1},
@@ -1735,6 +1762,7 @@ var _M = &RangeTable{
 		{0x10a3f, 0x10ae5, 166},
 		{0x10ae6, 0x10d24, 574},
 		{0x10d25, 0x10d27, 1},
+		{0x10eab, 0x10eac, 1},
 		{0x10f46, 0x10f50, 1},
 		{0x11000, 0x11002, 1},
 		{0x11038, 0x11046, 1},
@@ -1747,6 +1775,7 @@ var _M = &RangeTable{
 		{0x11181, 0x11182, 1},
 		{0x111b3, 0x111c0, 1},
 		{0x111c9, 0x111cc, 1},
+		{0x111ce, 0x111cf, 1},
 		{0x1122c, 0x11237, 1},
 		{0x1123e, 0x112df, 161},
 		{0x112e0, 0x112ea, 1},
@@ -1769,7 +1798,12 @@ var _M = &RangeTable{
 		{0x116ab, 0x116b7, 1},
 		{0x1171d, 0x1172b, 1},
 		{0x1182c, 0x1183a, 1},
-		{0x119d1, 0x119d7, 1},
+		{0x11930, 0x11935, 1},
+		{0x11937, 0x11938, 1},
+		{0x1193b, 0x1193e, 1},
+		{0x11940, 0x11942, 2},
+		{0x11943, 0x119d1, 142},
+		{0x119d2, 0x119d7, 1},
 		{0x119da, 0x119e0, 1},
 		{0x119e4, 0x11a01, 29},
 		{0x11a02, 0x11a0a, 1},
@@ -1796,8 +1830,10 @@ var _M = &RangeTable{
 		{0x16f4f, 0x16f51, 2},
 		{0x16f52, 0x16f87, 1},
 		{0x16f8f, 0x16f92, 1},
-		{0x1bc9d, 0x1bc9e, 1},
-		{0x1d165, 0x1d169, 1},
+		{0x16fe4, 0x16ff0, 12},
+		{0x16ff1, 0x1bc9d, 19628},
+		{0x1bc9e, 0x1d165, 5319},
+		{0x1d166, 0x1d169, 1},
 		{0x1d16d, 0x1d172, 1},
 		{0x1d17b, 0x1d182, 1},
 		{0x1d185, 0x1d18b, 1},
@@ -1929,7 +1965,8 @@ var _Mc = &RangeTable{
 		{0x11146, 0x11182, 60},
 		{0x111b3, 0x111b5, 1},
 		{0x111bf, 0x111c0, 1},
-		{0x1122c, 0x1122e, 1},
+		{0x111ce, 0x1122c, 94},
+		{0x1122d, 0x1122e, 1},
 		{0x11232, 0x11233, 1},
 		{0x11235, 0x112e0, 171},
 		{0x112e1, 0x112e2, 1},
@@ -1957,7 +1994,11 @@ var _Mc = &RangeTable{
 		{0x116b6, 0x11720, 106},
 		{0x11721, 0x11726, 5},
 		{0x1182c, 0x1182e, 1},
-		{0x11838, 0x119d1, 409},
+		{0x11838, 0x11930, 248},
+		{0x11931, 0x11935, 1},
+		{0x11937, 0x11938, 1},
+		{0x1193d, 0x11940, 3},
+		{0x11942, 0x119d1, 143},
 		{0x119d2, 0x119d3, 1},
 		{0x119dc, 0x119df, 1},
 		{0x119e4, 0x11a39, 85},
@@ -1970,6 +2011,7 @@ var _Mc = &RangeTable{
 		{0x11d96, 0x11ef5, 351},
 		{0x11ef6, 0x16f51, 20571},
 		{0x16f52, 0x16f87, 1},
+		{0x16ff0, 0x16ff1, 1},
 		{0x1d165, 0x1d166, 1},
 		{0x1d16d, 0x1d172, 1},
 	},
@@ -2038,12 +2080,12 @@ var _Mn = &RangeTable{
 		{0x0b01, 0x0b3c, 59},
 		{0x0b3f, 0x0b41, 2},
 		{0x0b42, 0x0b44, 1},
-		{0x0b4d, 0x0b56, 9},
-		{0x0b62, 0x0b63, 1},
-		{0x0b82, 0x0bc0, 62},
-		{0x0bcd, 0x0c00, 51},
-		{0x0c04, 0x0c3e, 58},
-		{0x0c3f, 0x0c40, 1},
+		{0x0b4d, 0x0b55, 8},
+		{0x0b56, 0x0b62, 12},
+		{0x0b63, 0x0b82, 31},
+		{0x0bc0, 0x0bcd, 13},
+		{0x0c00, 0x0c04, 4},
+		{0x0c3e, 0x0c40, 1},
 		{0x0c46, 0x0c48, 1},
 		{0x0c4a, 0x0c4d, 1},
 		{0x0c55, 0x0c56, 1},
@@ -2056,8 +2098,9 @@ var _Mn = &RangeTable{
 		{0x0d3b, 0x0d3c, 1},
 		{0x0d41, 0x0d44, 1},
 		{0x0d4d, 0x0d62, 21},
-		{0x0d63, 0x0dca, 103},
-		{0x0dd2, 0x0dd4, 1},
+		{0x0d63, 0x0d81, 30},
+		{0x0dca, 0x0dd2, 8},
+		{0x0dd3, 0x0dd4, 1},
 		{0x0dd6, 0x0e31, 91},
 		{0x0e34, 0x0e3a, 1},
 		{0x0e47, 0x0e4e, 1},
@@ -2107,6 +2150,7 @@ var _Mn = &RangeTable{
 		{0x1a73, 0x1a7c, 1},
 		{0x1a7f, 0x1ab0, 49},
 		{0x1ab1, 0x1abd, 1},
+		{0x1abf, 0x1ac0, 1},
 		{0x1b00, 0x1b03, 1},
 		{0x1b34, 0x1b36, 2},
 		{0x1b37, 0x1b3a, 1},
@@ -2142,9 +2186,9 @@ var _Mn = &RangeTable{
 		{0xa6f0, 0xa6f1, 1},
 		{0xa802, 0xa806, 4},
 		{0xa80b, 0xa825, 26},
-		{0xa826, 0xa8c4, 158},
-		{0xa8c5, 0xa8e0, 27},
-		{0xa8e1, 0xa8f1, 1},
+		{0xa826, 0xa82c, 6},
+		{0xa8c4, 0xa8c5, 1},
+		{0xa8e0, 0xa8f1, 1},
 		{0xa8ff, 0xa926, 39},
 		{0xa927, 0xa92d, 1},
 		{0xa947, 0xa951, 1},
@@ -2178,6 +2222,7 @@ var _Mn = &RangeTable{
 		{0x10a3f, 0x10ae5, 166},
 		{0x10ae6, 0x10d24, 574},
 		{0x10d25, 0x10d27, 1},
+		{0x10eab, 0x10eac, 1},
 		{0x10f46, 0x10f50, 1},
 		{0x11001, 0x11038, 55},
 		{0x11039, 0x11046, 1},
@@ -2191,7 +2236,8 @@ var _Mn = &RangeTable{
 		{0x11181, 0x111b6, 53},
 		{0x111b7, 0x111be, 1},
 		{0x111c9, 0x111cc, 1},
-		{0x1122f, 0x11231, 1},
+		{0x111cf, 0x1122f, 96},
+		{0x11230, 0x11231, 1},
 		{0x11234, 0x11236, 2},
 		{0x11237, 0x1123e, 7},
 		{0x112df, 0x112e3, 4},
@@ -2223,6 +2269,8 @@ var _Mn = &RangeTable{
 		{0x11727, 0x1172b, 1},
 		{0x1182f, 0x11837, 1},
 		{0x11839, 0x1183a, 1},
+		{0x1193b, 0x1193c, 1},
+		{0x1193e, 0x11943, 5},
 		{0x119d4, 0x119d7, 1},
 		{0x119da, 0x119db, 1},
 		{0x119e0, 0x11a01, 33},
@@ -2253,8 +2301,9 @@ var _Mn = &RangeTable{
 		{0x16b30, 0x16b36, 1},
 		{0x16f4f, 0x16f8f, 64},
 		{0x16f90, 0x16f92, 1},
-		{0x1bc9d, 0x1bc9e, 1},
-		{0x1d167, 0x1d169, 1},
+		{0x16fe4, 0x1bc9d, 19641},
+		{0x1bc9e, 0x1d167, 5321},
+		{0x1d168, 0x1d169, 1},
 		{0x1d17b, 0x1d182, 1},
 		{0x1d185, 0x1d18b, 1},
 		{0x1d1aa, 0x1d1ad, 1},
@@ -2375,6 +2424,7 @@ var _N = &RangeTable{
 		{0x10e60, 0x10e7e, 1},
 		{0x10f1d, 0x10f26, 1},
 		{0x10f51, 0x10f54, 1},
+		{0x10fc5, 0x10fcb, 1},
 		{0x11052, 0x1106f, 1},
 		{0x110f0, 0x110f9, 1},
 		{0x11136, 0x1113f, 1},
@@ -2387,6 +2437,7 @@ var _N = &RangeTable{
 		{0x116c0, 0x116c9, 1},
 		{0x11730, 0x1173b, 1},
 		{0x118e0, 0x118f2, 1},
+		{0x11950, 0x11959, 1},
 		{0x11c50, 0x11c6c, 1},
 		{0x11d50, 0x11d59, 1},
 		{0x11da0, 0x11da9, 1},
@@ -2409,6 +2460,7 @@ var _N = &RangeTable{
 		{0x1ed01, 0x1ed2d, 1},
 		{0x1ed2f, 0x1ed3d, 1},
 		{0x1f100, 0x1f10c, 1},
+		{0x1fbf0, 0x1fbf9, 1},
 	},
 	LatinOffset: 4,
 }
@@ -2467,6 +2519,7 @@ var _Nd = &RangeTable{
 		{0x116c0, 0x116c9, 1},
 		{0x11730, 0x11739, 1},
 		{0x118e0, 0x118e9, 1},
+		{0x11950, 0x11959, 1},
 		{0x11c50, 0x11c59, 1},
 		{0x11d50, 0x11d59, 1},
 		{0x11da0, 0x11da9, 1},
@@ -2476,6 +2529,7 @@ var _Nd = &RangeTable{
 		{0x1e140, 0x1e149, 1},
 		{0x1e2f0, 0x1e2f9, 1},
 		{0x1e950, 0x1e959, 1},
+		{0x1fbf0, 0x1fbf9, 1},
 	},
 	LatinOffset: 1,
 }
@@ -2554,6 +2608,7 @@ var _No = &RangeTable{
 		{0x10e60, 0x10e7e, 1},
 		{0x10f1d, 0x10f26, 1},
 		{0x10f51, 0x10f54, 1},
+		{0x10fc5, 0x10fcb, 1},
 		{0x11052, 0x11065, 1},
 		{0x111e1, 0x111f4, 1},
 		{0x1173a, 0x1173b, 1},
@@ -2655,7 +2710,8 @@ var _P = &RangeTable{
 		{0x2d70, 0x2e00, 144},
 		{0x2e01, 0x2e2e, 1},
 		{0x2e30, 0x2e4f, 1},
-		{0x3001, 0x3003, 1},
+		{0x2e52, 0x3001, 431},
+		{0x3002, 0x3003, 1},
 		{0x3008, 0x3011, 1},
 		{0x3014, 0x301f, 1},
 		{0x3030, 0x303d, 13},
@@ -2701,7 +2757,8 @@ var _P = &RangeTable{
 		{0x10af1, 0x10af6, 1},
 		{0x10b39, 0x10b3f, 1},
 		{0x10b99, 0x10b9c, 1},
-		{0x10f55, 0x10f59, 1},
+		{0x10ead, 0x10f55, 168},
+		{0x10f56, 0x10f59, 1},
 		{0x11047, 0x1104d, 1},
 		{0x110bb, 0x110bc, 1},
 		{0x110be, 0x110c1, 1},
@@ -2713,14 +2770,16 @@ var _P = &RangeTable{
 		{0x11238, 0x1123d, 1},
 		{0x112a9, 0x1144b, 418},
 		{0x1144c, 0x1144f, 1},
-		{0x1145b, 0x1145d, 2},
-		{0x114c6, 0x115c1, 251},
-		{0x115c2, 0x115d7, 1},
+		{0x1145a, 0x1145b, 1},
+		{0x1145d, 0x114c6, 105},
+		{0x115c1, 0x115d7, 1},
 		{0x11641, 0x11643, 1},
 		{0x11660, 0x1166c, 1},
 		{0x1173c, 0x1173e, 1},
-		{0x1183b, 0x119e2, 423},
-		{0x11a3f, 0x11a46, 1},
+		{0x1183b, 0x11944, 265},
+		{0x11945, 0x11946, 1},
+		{0x119e2, 0x11a3f, 93},
+		{0x11a40, 0x11a46, 1},
 		{0x11a9a, 0x11a9c, 1},
 		{0x11a9e, 0x11aa2, 1},
 		{0x11c41, 0x11c45, 1},
@@ -2764,6 +2823,9 @@ var _Pd = &RangeTable{
 		{0xfe58, 0xfe63, 11},
 		{0xff0d, 0xff0d, 1},
 	},
+	R32: []Range32{
+		{0x10ead, 0x10ead, 1},
+	},
 }
 
 var _Pe = &RangeTable{
@@ -2894,7 +2956,8 @@ var _Po = &RangeTable{
 		{0x2e3c, 0x2e3f, 1},
 		{0x2e41, 0x2e43, 2},
 		{0x2e44, 0x2e4f, 1},
-		{0x3001, 0x3003, 1},
+		{0x2e52, 0x3001, 431},
+		{0x3002, 0x3003, 1},
 		{0x303d, 0x30fb, 190},
 		{0xa4fe, 0xa4ff, 1},
 		{0xa60d, 0xa60f, 1},
@@ -2951,14 +3014,16 @@ var _Po = &RangeTable{
 		{0x11238, 0x1123d, 1},
 		{0x112a9, 0x1144b, 418},
 		{0x1144c, 0x1144f, 1},
-		{0x1145b, 0x1145d, 2},
-		{0x114c6, 0x115c1, 251},
-		{0x115c2, 0x115d7, 1},
+		{0x1145a, 0x1145b, 1},
+		{0x1145d, 0x114c6, 105},
+		{0x115c1, 0x115d7, 1},
 		{0x11641, 0x11643, 1},
 		{0x11660, 0x1166c, 1},
 		{0x1173c, 0x1173e, 1},
-		{0x1183b, 0x119e2, 423},
-		{0x11a3f, 0x11a46, 1},
+		{0x1183b, 0x11944, 265},
+		{0x11945, 0x11946, 1},
+		{0x119e2, 0x11a3f, 93},
+		{0x11a40, 0x11a46, 1},
 		{0x11a9a, 0x11a9c, 1},
 		{0x11a9e, 0x11aa2, 1},
 		{0x11c41, 0x11c45, 1},
@@ -3094,8 +3159,9 @@ var _S = &RangeTable{
 		{0x29dc, 0x29fb, 1},
 		{0x29fe, 0x2b73, 1},
 		{0x2b76, 0x2b95, 1},
-		{0x2b98, 0x2bff, 1},
+		{0x2b97, 0x2bff, 1},
 		{0x2ce5, 0x2cea, 1},
+		{0x2e50, 0x2e51, 1},
 		{0x2e80, 0x2e99, 1},
 		{0x2e9b, 0x2ef3, 1},
 		{0x2f00, 0x2fd5, 1},
@@ -3113,8 +3179,7 @@ var _S = &RangeTable{
 		{0x3250, 0x3260, 16},
 		{0x3261, 0x327f, 1},
 		{0x328a, 0x32b0, 1},
-		{0x32c0, 0x32fe, 1},
-		{0x3300, 0x33ff, 1},
+		{0x32c0, 0x33ff, 1},
 		{0x4dc0, 0x4dff, 1},
 		{0xa490, 0xa4c6, 1},
 		{0xa700, 0xa716, 1},
@@ -3123,7 +3188,8 @@ var _S = &RangeTable{
 		{0xa828, 0xa82b, 1},
 		{0xa836, 0xa839, 1},
 		{0xaa77, 0xaa79, 1},
-		{0xab5b, 0xfb29, 20430},
+		{0xab5b, 0xab6a, 15},
+		{0xab6b, 0xfb29, 20414},
 		{0xfbb2, 0xfbc1, 1},
 		{0xfdfc, 0xfdfd, 1},
 		{0xfe62, 0xfe64, 2},
@@ -3141,7 +3207,7 @@ var _S = &RangeTable{
 		{0x10137, 0x1013f, 1},
 		{0x10179, 0x10189, 1},
 		{0x1018c, 0x1018e, 1},
-		{0x10190, 0x1019b, 1},
+		{0x10190, 0x1019c, 1},
 		{0x101a0, 0x101d0, 48},
 		{0x101d1, 0x101fc, 1},
 		{0x10877, 0x10878, 1},
@@ -3179,16 +3245,15 @@ var _S = &RangeTable{
 		{0x1f0b1, 0x1f0bf, 1},
 		{0x1f0c1, 0x1f0cf, 1},
 		{0x1f0d1, 0x1f0f5, 1},
-		{0x1f110, 0x1f16c, 1},
-		{0x1f170, 0x1f1ac, 1},
+		{0x1f10d, 0x1f1ad, 1},
 		{0x1f1e6, 0x1f202, 1},
 		{0x1f210, 0x1f23b, 1},
 		{0x1f240, 0x1f248, 1},
 		{0x1f250, 0x1f251, 1},
 		{0x1f260, 0x1f265, 1},
-		{0x1f300, 0x1f6d5, 1},
+		{0x1f300, 0x1f6d7, 1},
 		{0x1f6e0, 0x1f6ec, 1},
-		{0x1f6f0, 0x1f6fa, 1},
+		{0x1f6f0, 0x1f6fc, 1},
 		{0x1f700, 0x1f773, 1},
 		{0x1f780, 0x1f7d8, 1},
 		{0x1f7e0, 0x1f7eb, 1},
@@ -3197,18 +3262,20 @@ var _S = &RangeTable{
 		{0x1f850, 0x1f859, 1},
 		{0x1f860, 0x1f887, 1},
 		{0x1f890, 0x1f8ad, 1},
-		{0x1f900, 0x1f90b, 1},
-		{0x1f90d, 0x1f971, 1},
-		{0x1f973, 0x1f976, 1},
-		{0x1f97a, 0x1f9a2, 1},
-		{0x1f9a5, 0x1f9aa, 1},
-		{0x1f9ae, 0x1f9ca, 1},
+		{0x1f8b0, 0x1f8b1, 1},
+		{0x1f900, 0x1f978, 1},
+		{0x1f97a, 0x1f9cb, 1},
 		{0x1f9cd, 0x1fa53, 1},
 		{0x1fa60, 0x1fa6d, 1},
-		{0x1fa70, 0x1fa73, 1},
+		{0x1fa70, 0x1fa74, 1},
 		{0x1fa78, 0x1fa7a, 1},
-		{0x1fa80, 0x1fa82, 1},
-		{0x1fa90, 0x1fa95, 1},
+		{0x1fa80, 0x1fa86, 1},
+		{0x1fa90, 0x1faa8, 1},
+		{0x1fab0, 0x1fab6, 1},
+		{0x1fac0, 0x1fac2, 1},
+		{0x1fad0, 0x1fad6, 1},
+		{0x1fb00, 0x1fb92, 1},
+		{0x1fb94, 0x1fbca, 1},
 	},
 	LatinOffset: 10,
 }
@@ -3257,7 +3324,8 @@ var _Sk = &RangeTable{
 		{0xa700, 0xa716, 1},
 		{0xa720, 0xa721, 1},
 		{0xa789, 0xa78a, 1},
-		{0xab5b, 0xfbb2, 20567},
+		{0xab5b, 0xab6a, 15},
+		{0xab6b, 0xfbb2, 20551},
 		{0xfbb3, 0xfbc1, 1},
 		{0xff3e, 0xff40, 2},
 		{0xffe3, 0xffe3, 1},
@@ -3394,8 +3462,9 @@ var _So = &RangeTable{
 		{0x2b45, 0x2b46, 1},
 		{0x2b4d, 0x2b73, 1},
 		{0x2b76, 0x2b95, 1},
-		{0x2b98, 0x2bff, 1},
+		{0x2b97, 0x2bff, 1},
 		{0x2ce5, 0x2cea, 1},
+		{0x2e50, 0x2e51, 1},
 		{0x2e80, 0x2e99, 1},
 		{0x2e9b, 0x2ef3, 1},
 		{0x2f00, 0x2fd5, 1},
@@ -3412,8 +3481,7 @@ var _So = &RangeTable{
 		{0x3250, 0x3260, 16},
 		{0x3261, 0x327f, 1},
 		{0x328a, 0x32b0, 1},
-		{0x32c0, 0x32fe, 1},
-		{0x3300, 0x33ff, 1},
+		{0x32c0, 0x33ff, 1},
 		{0x4dc0, 0x4dff, 1},
 		{0xa490, 0xa4c6, 1},
 		{0xa828, 0xa82b, 1},
@@ -3429,7 +3497,7 @@ var _So = &RangeTable{
 		{0x10137, 0x1013f, 1},
 		{0x10179, 0x10189, 1},
 		{0x1018c, 0x1018e, 1},
-		{0x10190, 0x1019b, 1},
+		{0x10190, 0x1019c, 1},
 		{0x101a0, 0x101d0, 48},
 		{0x101d1, 0x101fc, 1},
 		{0x10877, 0x10878, 1},
@@ -3461,17 +3529,16 @@ var _So = &RangeTable{
 		{0x1f0b1, 0x1f0bf, 1},
 		{0x1f0c1, 0x1f0cf, 1},
 		{0x1f0d1, 0x1f0f5, 1},
-		{0x1f110, 0x1f16c, 1},
-		{0x1f170, 0x1f1ac, 1},
+		{0x1f10d, 0x1f1ad, 1},
 		{0x1f1e6, 0x1f202, 1},
 		{0x1f210, 0x1f23b, 1},
 		{0x1f240, 0x1f248, 1},
 		{0x1f250, 0x1f251, 1},
 		{0x1f260, 0x1f265, 1},
 		{0x1f300, 0x1f3fa, 1},
-		{0x1f400, 0x1f6d5, 1},
+		{0x1f400, 0x1f6d7, 1},
 		{0x1f6e0, 0x1f6ec, 1},
-		{0x1f6f0, 0x1f6fa, 1},
+		{0x1f6f0, 0x1f6fc, 1},
 		{0x1f700, 0x1f773, 1},
 		{0x1f780, 0x1f7d8, 1},
 		{0x1f7e0, 0x1f7eb, 1},
@@ -3480,18 +3547,20 @@ var _So = &RangeTable{
 		{0x1f850, 0x1f859, 1},
 		{0x1f860, 0x1f887, 1},
 		{0x1f890, 0x1f8ad, 1},
-		{0x1f900, 0x1f90b, 1},
-		{0x1f90d, 0x1f971, 1},
-		{0x1f973, 0x1f976, 1},
-		{0x1f97a, 0x1f9a2, 1},
-		{0x1f9a5, 0x1f9aa, 1},
-		{0x1f9ae, 0x1f9ca, 1},
+		{0x1f8b0, 0x1f8b1, 1},
+		{0x1f900, 0x1f978, 1},
+		{0x1f97a, 0x1f9cb, 1},
 		{0x1f9cd, 0x1fa53, 1},
 		{0x1fa60, 0x1fa6d, 1},
-		{0x1fa70, 0x1fa73, 1},
+		{0x1fa70, 0x1fa74, 1},
 		{0x1fa78, 0x1fa7a, 1},
-		{0x1fa80, 0x1fa82, 1},
-		{0x1fa90, 0x1fa95, 1},
+		{0x1fa80, 0x1fa86, 1},
+		{0x1fa90, 0x1faa8, 1},
+		{0x1fab0, 0x1fab6, 1},
+		{0x1fac0, 0x1fac2, 1},
+		{0x1fad0, 0x1fad6, 1},
+		{0x1fb00, 0x1fb92, 1},
+		{0x1fb94, 0x1fbca, 1},
 	},
 	LatinOffset: 2,
 }
@@ -3607,6 +3676,7 @@ var Scripts = map[string]*RangeTable{
 	"Chakma":                 Chakma,
 	"Cham":                   Cham,
 	"Cherokee":               Cherokee,
+	"Chorasmian":             Chorasmian,
 	"Common":                 Common,
 	"Coptic":                 Coptic,
 	"Cuneiform":              Cuneiform,
@@ -3614,6 +3684,7 @@ var Scripts = map[string]*RangeTable{
 	"Cyrillic":               Cyrillic,
 	"Deseret":                Deseret,
 	"Devanagari":             Devanagari,
+	"Dives_Akuru":            Dives_Akuru,
 	"Dogra":                  Dogra,
 	"Duployan":               Duployan,
 	"Egyptian_Hieroglyphs":   Egyptian_Hieroglyphs,
@@ -3645,6 +3716,7 @@ var Scripts = map[string]*RangeTable{
 	"Katakana":               Katakana,
 	"Kayah_Li":               Kayah_Li,
 	"Kharoshthi":             Kharoshthi,
+	"Khitan_Small_Script":    Khitan_Small_Script,
 	"Khmer":                  Khmer,
 	"Khojki":                 Khojki,
 	"Khudawadi":              Khudawadi,
@@ -3734,6 +3806,7 @@ var Scripts = map[string]*RangeTable{
 	"Vai":                    Vai,
 	"Wancho":                 Wancho,
 	"Warang_Citi":            Warang_Citi,
+	"Yezidi":                 Yezidi,
 	"Yi":                     Yi,
 	"Zanabazar_Square":       Zanabazar_Square,
 }
@@ -3776,7 +3849,7 @@ var _Arabic = &RangeTable{
 		{0x06de, 0x06ff, 1},
 		{0x0750, 0x077f, 1},
 		{0x08a0, 0x08b4, 1},
-		{0x08b6, 0x08bd, 1},
+		{0x08b6, 0x08c7, 1},
 		{0x08d3, 0x08e1, 1},
 		{0x08e3, 0x08ff, 1},
 		{0xfb50, 0xfbc1, 1},
@@ -3820,9 +3893,8 @@ var _Arabic = &RangeTable{
 var _Armenian = &RangeTable{
 	R16: []Range16{
 		{0x0531, 0x0556, 1},
-		{0x0559, 0x0588, 1},
-		{0x058a, 0x058d, 3},
-		{0x058e, 0x058f, 1},
+		{0x0559, 0x058a, 1},
+		{0x058d, 0x058f, 1},
 		{0xfb13, 0xfb17, 1},
 	},
 }
@@ -3899,7 +3971,7 @@ var _Bopomofo = &RangeTable{
 	R16: []Range16{
 		{0x02ea, 0x02eb, 1},
 		{0x3105, 0x312f, 1},
-		{0x31a0, 0x31ba, 1},
+		{0x31a0, 0x31bf, 1},
 	},
 }
 
@@ -3957,7 +4029,7 @@ var _Chakma = &RangeTable{
 	R16: []Range16{},
 	R32: []Range32{
 		{0x11100, 0x11134, 1},
-		{0x11136, 0x11146, 1},
+		{0x11136, 0x11147, 1},
 	},
 }
 
@@ -3978,6 +4050,13 @@ var _Cherokee = &RangeTable{
 	},
 }
 
+var _Chorasmian = &RangeTable{
+	R16: []Range16{},
+	R32: []Range32{
+		{0x10fb0, 0x10fcb, 1},
+	},
+}
+
 var _Common = &RangeTable{
 	R16: []Range16{
 		{0x0000, 0x0040, 1},
@@ -3991,13 +4070,12 @@ var _Common = &RangeTable{
 		{0x02ec, 0x02ff, 1},
 		{0x0374, 0x037e, 10},
 		{0x0385, 0x0387, 2},
-		{0x0589, 0x0605, 124},
-		{0x060c, 0x061b, 15},
-		{0x061f, 0x0640, 33},
-		{0x06dd, 0x08e2, 517},
-		{0x0964, 0x0965, 1},
-		{0x0e3f, 0x0fd5, 406},
-		{0x0fd6, 0x0fd8, 1},
+		{0x0605, 0x060c, 7},
+		{0x061b, 0x061f, 4},
+		{0x0640, 0x06dd, 157},
+		{0x08e2, 0x0964, 130},
+		{0x0965, 0x0e3f, 1242},
+		{0x0fd5, 0x0fd8, 1},
 		{0x10fb, 0x16eb, 1520},
 		{0x16ec, 0x16ed, 1},
 		{0x1735, 0x1736, 1},
@@ -4025,8 +4103,8 @@ var _Common = &RangeTable{
 		{0x2460, 0x27ff, 1},
 		{0x2900, 0x2b73, 1},
 		{0x2b76, 0x2b95, 1},
-		{0x2b98, 0x2bff, 1},
-		{0x2e00, 0x2e4f, 1},
+		{0x2b97, 0x2bff, 1},
+		{0x2e00, 0x2e52, 1},
 		{0x2ff0, 0x2ffb, 1},
 		{0x3000, 0x3004, 1},
 		{0x3006, 0x3008, 2},
@@ -4040,13 +4118,15 @@ var _Common = &RangeTable{
 		{0x31c0, 0x31e3, 1},
 		{0x3220, 0x325f, 1},
 		{0x327f, 0x32cf, 1},
-		{0x3358, 0x33ff, 1},
+		{0x32ff, 0x3358, 89},
+		{0x3359, 0x33ff, 1},
 		{0x4dc0, 0x4dff, 1},
 		{0xa700, 0xa721, 1},
 		{0xa788, 0xa78a, 1},
 		{0xa830, 0xa839, 1},
 		{0xa92e, 0xa9cf, 161},
-		{0xab5b, 0xfd3e, 20963},
+		{0xab5b, 0xab6a, 15},
+		{0xab6b, 0xfd3e, 20947},
 		{0xfd3f, 0xfe10, 209},
 		{0xfe11, 0xfe19, 1},
 		{0xfe30, 0xfe52, 1},
@@ -4066,7 +4146,7 @@ var _Common = &RangeTable{
 		{0x10100, 0x10102, 1},
 		{0x10107, 0x10133, 1},
 		{0x10137, 0x1013f, 1},
-		{0x10190, 0x1019b, 1},
+		{0x10190, 0x1019c, 1},
 		{0x101d0, 0x101fc, 1},
 		{0x102e1, 0x102fb, 1},
 		{0x16fe2, 0x16fe3, 1},
@@ -4110,18 +4190,16 @@ var _Common = &RangeTable{
 		{0x1f0b1, 0x1f0bf, 1},
 		{0x1f0c1, 0x1f0cf, 1},
 		{0x1f0d1, 0x1f0f5, 1},
-		{0x1f100, 0x1f10c, 1},
-		{0x1f110, 0x1f16c, 1},
-		{0x1f170, 0x1f1ac, 1},
+		{0x1f100, 0x1f1ad, 1},
 		{0x1f1e6, 0x1f1ff, 1},
 		{0x1f201, 0x1f202, 1},
 		{0x1f210, 0x1f23b, 1},
 		{0x1f240, 0x1f248, 1},
 		{0x1f250, 0x1f251, 1},
 		{0x1f260, 0x1f265, 1},
-		{0x1f300, 0x1f6d5, 1},
+		{0x1f300, 0x1f6d7, 1},
 		{0x1f6e0, 0x1f6ec, 1},
-		{0x1f6f0, 0x1f6fa, 1},
+		{0x1f6f0, 0x1f6fc, 1},
 		{0x1f700, 0x1f773, 1},
 		{0x1f780, 0x1f7d8, 1},
 		{0x1f7e0, 0x1f7eb, 1},
@@ -4130,18 +4208,21 @@ var _Common = &RangeTable{
 		{0x1f850, 0x1f859, 1},
 		{0x1f860, 0x1f887, 1},
 		{0x1f890, 0x1f8ad, 1},
-		{0x1f900, 0x1f90b, 1},
-		{0x1f90d, 0x1f971, 1},
-		{0x1f973, 0x1f976, 1},
-		{0x1f97a, 0x1f9a2, 1},
-		{0x1f9a5, 0x1f9aa, 1},
-		{0x1f9ae, 0x1f9ca, 1},
+		{0x1f8b0, 0x1f8b1, 1},
+		{0x1f900, 0x1f978, 1},
+		{0x1f97a, 0x1f9cb, 1},
 		{0x1f9cd, 0x1fa53, 1},
 		{0x1fa60, 0x1fa6d, 1},
-		{0x1fa70, 0x1fa73, 1},
+		{0x1fa70, 0x1fa74, 1},
 		{0x1fa78, 0x1fa7a, 1},
-		{0x1fa80, 0x1fa82, 1},
-		{0x1fa90, 0x1fa95, 1},
+		{0x1fa80, 0x1fa86, 1},
+		{0x1fa90, 0x1faa8, 1},
+		{0x1fab0, 0x1fab6, 1},
+		{0x1fac0, 0x1fac2, 1},
+		{0x1fad0, 0x1fad6, 1},
+		{0x1fb00, 0x1fb92, 1},
+		{0x1fb94, 0x1fbca, 1},
+		{0x1fbf0, 0x1fbf9, 1},
 		{0xe0001, 0xe0020, 31},
 		{0xe0021, 0xe007f, 1},
 	},
@@ -4205,6 +4286,20 @@ var _Devanagari = &RangeTable{
 	},
 }
 
+var _Dives_Akuru = &RangeTable{
+	R16: []Range16{},
+	R32: []Range32{
+		{0x11900, 0x11906, 1},
+		{0x11909, 0x1190c, 3},
+		{0x1190d, 0x11913, 1},
+		{0x11915, 0x11916, 1},
+		{0x11918, 0x11935, 1},
+		{0x11937, 0x11938, 1},
+		{0x1193b, 0x11946, 1},
+		{0x11950, 0x11959, 1},
+	},
+}
+
 var _Dogra = &RangeTable{
 	R16: []Range16{},
 	R32: []Range32{
@@ -4435,18 +4530,20 @@ var _Han = &RangeTable{
 		{0x3005, 0x3007, 2},
 		{0x3021, 0x3029, 1},
 		{0x3038, 0x303b, 1},
-		{0x3400, 0x4db5, 1},
-		{0x4e00, 0x9fef, 1},
+		{0x3400, 0x4dbf, 1},
+		{0x4e00, 0x9ffc, 1},
 		{0xf900, 0xfa6d, 1},
 		{0xfa70, 0xfad9, 1},
 	},
 	R32: []Range32{
-		{0x20000, 0x2a6d6, 1},
+		{0x16ff0, 0x16ff1, 1},
+		{0x20000, 0x2a6dd, 1},
 		{0x2a700, 0x2b734, 1},
 		{0x2b740, 0x2b81d, 1},
 		{0x2b820, 0x2cea1, 1},
 		{0x2ceb0, 0x2ebe0, 1},
 		{0x2f800, 0x2fa1d, 1},
+		{0x30000, 0x3134a, 1},
 	},
 }
 
@@ -4533,7 +4630,7 @@ var _Inherited = &RangeTable{
 		{0x064b, 0x0655, 1},
 		{0x0670, 0x0951, 737},
 		{0x0952, 0x0954, 1},
-		{0x1ab0, 0x1abe, 1},
+		{0x1ab0, 0x1ac0, 1},
 		{0x1cd0, 0x1cd2, 1},
 		{0x1cd4, 0x1ce0, 1},
 		{0x1ce2, 0x1ce8, 1},
@@ -4646,6 +4743,14 @@ var _Kharoshthi = &RangeTable{
 	},
 }
 
+var _Khitan_Small_Script = &RangeTable{
+	R16: []Range16{},
+	R32: []Range32{
+		{0x16fe4, 0x18b00, 6940},
+		{0x18b01, 0x18cd5, 1},
+	},
+}
+
 var _Khmer = &RangeTable{
 	R16: []Range16{
 		{0x1780, 0x17dd, 1},
@@ -4710,11 +4815,11 @@ var _Latin = &RangeTable{
 		{0x2c60, 0x2c7f, 1},
 		{0xa722, 0xa787, 1},
 		{0xa78b, 0xa7bf, 1},
-		{0xa7c2, 0xa7c6, 1},
-		{0xa7f7, 0xa7ff, 1},
+		{0xa7c2, 0xa7ca, 1},
+		{0xa7f5, 0xa7ff, 1},
 		{0xab30, 0xab5a, 1},
 		{0xab5c, 0xab64, 1},
-		{0xab66, 0xab67, 1},
+		{0xab66, 0xab69, 1},
 		{0xfb00, 0xfb06, 1},
 		{0xff21, 0xff3a, 1},
 		{0xff41, 0xff5a, 1},
@@ -4766,6 +4871,9 @@ var _Lisu = &RangeTable{
 	R16: []Range16{
 		{0xa4d0, 0xa4ff, 1},
 	},
+	R32: []Range32{
+		{0x11fb0, 0x11fb0, 1},
+	},
 }
 
 var _Lycian = &RangeTable{
@@ -4799,8 +4907,7 @@ var _Makasar = &RangeTable{
 
 var _Malayalam = &RangeTable{
 	R16: []Range16{
-		{0x0d00, 0x0d03, 1},
-		{0x0d05, 0x0d0c, 1},
+		{0x0d00, 0x0d0c, 1},
 		{0x0d0e, 0x0d10, 1},
 		{0x0d12, 0x0d44, 1},
 		{0x0d46, 0x0d48, 1},
@@ -4974,9 +5081,8 @@ var _New_Tai_Lue = &RangeTable{
 var _Newa = &RangeTable{
 	R16: []Range16{},
 	R32: []Range32{
-		{0x11400, 0x11459, 1},
-		{0x1145b, 0x1145d, 2},
-		{0x1145e, 0x1145f, 1},
+		{0x11400, 0x1145b, 1},
+		{0x1145d, 0x11461, 1},
 	},
 }
 
@@ -5089,7 +5195,7 @@ var _Oriya = &RangeTable{
 		{0x0b3c, 0x0b44, 1},
 		{0x0b47, 0x0b48, 1},
 		{0x0b4b, 0x0b4d, 1},
-		{0x0b56, 0x0b57, 1},
+		{0x0b55, 0x0b57, 1},
 		{0x0b5c, 0x0b5d, 1},
 		{0x0b5f, 0x0b63, 1},
 		{0x0b66, 0x0b77, 1},
@@ -5191,8 +5297,7 @@ var _Saurashtra = &RangeTable{
 var _Sharada = &RangeTable{
 	R16: []Range16{},
 	R32: []Range32{
-		{0x11180, 0x111cd, 1},
-		{0x111d0, 0x111df, 1},
+		{0x11180, 0x111df, 1},
 	},
 }
 
@@ -5222,7 +5327,7 @@ var _SignWriting = &RangeTable{
 
 var _Sinhala = &RangeTable{
 	R16: []Range16{
-		{0x0d82, 0x0d83, 1},
+		{0x0d81, 0x0d83, 1},
 		{0x0d85, 0x0d96, 1},
 		{0x0d9a, 0x0db1, 1},
 		{0x0db3, 0x0dbb, 1},
@@ -5271,7 +5376,7 @@ var _Sundanese = &RangeTable{
 
 var _Syloti_Nagri = &RangeTable{
 	R16: []Range16{
-		{0xa800, 0xa82b, 1},
+		{0xa800, 0xa82c, 1},
 	},
 }
 
@@ -5360,7 +5465,8 @@ var _Tangut = &RangeTable{
 	R32: []Range32{
 		{0x16fe0, 0x17000, 32},
 		{0x17001, 0x187f7, 1},
-		{0x18800, 0x18af2, 1},
+		{0x18800, 0x18aff, 1},
+		{0x18d00, 0x18d08, 1},
 	},
 }
 
@@ -5452,6 +5558,15 @@ var _Warang_Citi = &RangeTable{
 	},
 }
 
+var _Yezidi = &RangeTable{
+	R16: []Range16{},
+	R32: []Range32{
+		{0x10e80, 0x10ea9, 1},
+		{0x10eab, 0x10ead, 1},
+		{0x10eb0, 0x10eb1, 1},
+	},
+}
+
 var _Yi = &RangeTable{
 	R16: []Range16{
 		{0xa000, 0xa48c, 1},
@@ -5491,6 +5606,7 @@ var (
 	Chakma                 = _Chakma                 // Chakma is the set of Unicode characters in script Chakma.
 	Cham                   = _Cham                   // Cham is the set of Unicode characters in script Cham.
 	Cherokee               = _Cherokee               // Cherokee is the set of Unicode characters in script Cherokee.
+	Chorasmian             = _Chorasmian             // Chorasmian is the set of Unicode characters in script Chorasmian.
 	Common                 = _Common                 // Common is the set of Unicode characters in script Common.
 	Coptic                 = _Coptic                 // Coptic is the set of Unicode characters in script Coptic.
 	Cuneiform              = _Cuneiform              // Cuneiform is the set of Unicode characters in script Cuneiform.
@@ -5498,6 +5614,7 @@ var (
 	Cyrillic               = _Cyrillic               // Cyrillic is the set of Unicode characters in script Cyrillic.
 	Deseret                = _Deseret                // Deseret is the set of Unicode characters in script Deseret.
 	Devanagari             = _Devanagari             // Devanagari is the set of Unicode characters in script Devanagari.
+	Dives_Akuru            = _Dives_Akuru            // Dives_Akuru is the set of Unicode characters in script Dives_Akuru.
 	Dogra                  = _Dogra                  // Dogra is the set of Unicode characters in script Dogra.
 	Duployan               = _Duployan               // Duployan is the set of Unicode characters in script Duployan.
 	Egyptian_Hieroglyphs   = _Egyptian_Hieroglyphs   // Egyptian_Hieroglyphs is the set of Unicode characters in script Egyptian_Hieroglyphs.
@@ -5529,6 +5646,7 @@ var (
 	Katakana               = _Katakana               // Katakana is the set of Unicode characters in script Katakana.
 	Kayah_Li               = _Kayah_Li               // Kayah_Li is the set of Unicode characters in script Kayah_Li.
 	Kharoshthi             = _Kharoshthi             // Kharoshthi is the set of Unicode characters in script Kharoshthi.
+	Khitan_Small_Script    = _Khitan_Small_Script    // Khitan_Small_Script is the set of Unicode characters in script Khitan_Small_Script.
 	Khmer                  = _Khmer                  // Khmer is the set of Unicode characters in script Khmer.
 	Khojki                 = _Khojki                 // Khojki is the set of Unicode characters in script Khojki.
 	Khudawadi              = _Khudawadi              // Khudawadi is the set of Unicode characters in script Khudawadi.
@@ -5618,6 +5736,7 @@ var (
 	Vai                    = _Vai                    // Vai is the set of Unicode characters in script Vai.
 	Wancho                 = _Wancho                 // Wancho is the set of Unicode characters in script Wancho.
 	Warang_Citi            = _Warang_Citi            // Warang_Citi is the set of Unicode characters in script Warang_Citi.
+	Yezidi                 = _Yezidi                 // Yezidi is the set of Unicode characters in script Yezidi.
 	Yi                     = _Yi                     // Yi is the set of Unicode characters in script Yi.
 	Zanabazar_Square       = _Zanabazar_Square       // Zanabazar_Square is the set of Unicode characters in script Zanabazar_Square.
 )
@@ -5695,6 +5814,9 @@ var _Dash = &RangeTable{
 		{0xfe58, 0xfe63, 11},
 		{0xff0d, 0xff0d, 1},
 	},
+	R32: []Range32{
+		{0x10ead, 0x10ead, 1},
+	},
 }
 
 var _Deprecated = &RangeTable{
@@ -5746,10 +5868,11 @@ var _Diacritic = &RangeTable{
 		{0x0acd, 0x0afd, 48},
 		{0x0afe, 0x0aff, 1},
 		{0x0b3c, 0x0b4d, 17},
-		{0x0bcd, 0x0c4d, 128},
-		{0x0cbc, 0x0ccd, 17},
-		{0x0d3b, 0x0d3c, 1},
-		{0x0d4d, 0x0e47, 125},
+		{0x0b55, 0x0bcd, 120},
+		{0x0c4d, 0x0cbc, 111},
+		{0x0ccd, 0x0d3b, 110},
+		{0x0d3c, 0x0d4d, 17},
+		{0x0dca, 0x0e47, 125},
 		{0x0e48, 0x0e4c, 1},
 		{0x0e4e, 0x0eba, 108},
 		{0x0ec8, 0x0ecc, 1},
@@ -5811,6 +5934,7 @@ var _Diacritic = &RangeTable{
 		{0xaabf, 0xaac2, 1},
 		{0xaaf6, 0xab5b, 101},
 		{0xab5c, 0xab5f, 1},
+		{0xab69, 0xab6b, 1},
 		{0xabec, 0xabed, 1},
 		{0xfb1e, 0xfe20, 770},
 		{0xfe21, 0xfe2f, 1},
@@ -5838,14 +5962,16 @@ var _Diacritic = &RangeTable{
 		{0x1163f, 0x116b6, 119},
 		{0x116b7, 0x1172b, 116},
 		{0x11839, 0x1183a, 1},
-		{0x119e0, 0x11a34, 84},
-		{0x11a47, 0x11a99, 82},
-		{0x11c3f, 0x11d42, 259},
-		{0x11d44, 0x11d45, 1},
-		{0x11d97, 0x16af0, 19801},
-		{0x16af1, 0x16af4, 1},
+		{0x1193d, 0x1193e, 1},
+		{0x11943, 0x119e0, 157},
+		{0x11a34, 0x11a47, 19},
+		{0x11a99, 0x11c3f, 422},
+		{0x11d42, 0x11d44, 2},
+		{0x11d45, 0x11d97, 82},
+		{0x16af0, 0x16af4, 1},
 		{0x16b30, 0x16b36, 1},
 		{0x16f8f, 0x16f9f, 1},
+		{0x16ff0, 0x16ff1, 1},
 		{0x1d167, 0x1d169, 1},
 		{0x1d16d, 0x1d172, 1},
 		{0x1d17b, 0x1d182, 1},
@@ -5864,12 +5990,12 @@ var _Extender = &RangeTable{
 	R16: []Range16{
 		{0x00b7, 0x02d0, 537},
 		{0x02d1, 0x0640, 879},
-		{0x07fa, 0x0e46, 1612},
-		{0x0ec6, 0x180a, 2372},
-		{0x1843, 0x1aa7, 612},
-		{0x1c36, 0x1c7b, 69},
-		{0x3005, 0x3031, 44},
-		{0x3032, 0x3035, 1},
+		{0x07fa, 0x0b55, 859},
+		{0x0e46, 0x0ec6, 128},
+		{0x180a, 0x1843, 57},
+		{0x1aa7, 0x1c36, 399},
+		{0x1c7b, 0x3005, 5002},
+		{0x3031, 0x3035, 1},
 		{0x309d, 0x309e, 1},
 		{0x30fc, 0x30fe, 1},
 		{0xa015, 0xa60c, 1527},
@@ -5931,21 +6057,24 @@ var _Ideographic = &RangeTable{
 		{0x3006, 0x3007, 1},
 		{0x3021, 0x3029, 1},
 		{0x3038, 0x303a, 1},
-		{0x3400, 0x4db5, 1},
-		{0x4e00, 0x9fef, 1},
+		{0x3400, 0x4dbf, 1},
+		{0x4e00, 0x9ffc, 1},
 		{0xf900, 0xfa6d, 1},
 		{0xfa70, 0xfad9, 1},
 	},
 	R32: []Range32{
-		{0x17000, 0x187f7, 1},
-		{0x18800, 0x18af2, 1},
+		{0x16fe4, 0x17000, 28},
+		{0x17001, 0x187f7, 1},
+		{0x18800, 0x18cd5, 1},
+		{0x18d00, 0x18d08, 1},
 		{0x1b170, 0x1b2fb, 1},
-		{0x20000, 0x2a6d6, 1},
+		{0x20000, 0x2a6dd, 1},
 		{0x2a700, 0x2b734, 1},
 		{0x2b740, 0x2b81d, 1},
 		{0x2b820, 0x2cea1, 1},
 		{0x2ceb0, 0x2ebe0, 1},
 		{0x2f800, 0x2fa1d, 1},
+		{0x30000, 0x3134a, 1},
 	},
 }
 
@@ -6066,9 +6195,9 @@ var _Other_Alphabetic = &RangeTable{
 		{0x0d46, 0x0d48, 1},
 		{0x0d4a, 0x0d4c, 1},
 		{0x0d57, 0x0d62, 11},
-		{0x0d63, 0x0d82, 31},
-		{0x0d83, 0x0dcf, 76},
-		{0x0dd0, 0x0dd4, 1},
+		{0x0d63, 0x0d81, 30},
+		{0x0d82, 0x0d83, 1},
+		{0x0dcf, 0x0dd4, 1},
 		{0x0dd6, 0x0dd8, 2},
 		{0x0dd9, 0x0ddf, 1},
 		{0x0df2, 0x0df3, 1},
@@ -6104,6 +6233,7 @@ var _Other_Alphabetic = &RangeTable{
 		{0x1a17, 0x1a1b, 1},
 		{0x1a55, 0x1a5e, 1},
 		{0x1a61, 0x1a74, 1},
+		{0x1abf, 0x1ac0, 1},
 		{0x1b00, 0x1b04, 1},
 		{0x1b35, 0x1b43, 1},
 		{0x1b80, 0x1b82, 1},
@@ -6145,6 +6275,7 @@ var _Other_Alphabetic = &RangeTable{
 		{0x10a05, 0x10a06, 1},
 		{0x10a0c, 0x10a0f, 1},
 		{0x10d24, 0x10d27, 1},
+		{0x10eab, 0x10eac, 1},
 		{0x11000, 0x11002, 1},
 		{0x11038, 0x11045, 1},
 		{0x11082, 0x110b0, 46},
@@ -6154,6 +6285,7 @@ var _Other_Alphabetic = &RangeTable{
 		{0x11145, 0x11146, 1},
 		{0x11180, 0x11182, 1},
 		{0x111b3, 0x111bf, 1},
+		{0x111ce, 0x111cf, 1},
 		{0x1122c, 0x11234, 1},
 		{0x11237, 0x1123e, 7},
 		{0x112df, 0x112e8, 1},
@@ -6174,6 +6306,10 @@ var _Other_Alphabetic = &RangeTable{
 		{0x116ac, 0x116b5, 1},
 		{0x1171d, 0x1172a, 1},
 		{0x1182c, 0x11838, 1},
+		{0x11930, 0x11935, 1},
+		{0x11937, 0x11938, 1},
+		{0x1193b, 0x1193c, 1},
+		{0x11940, 0x11942, 2},
 		{0x119d1, 0x119d7, 1},
 		{0x119da, 0x119df, 1},
 		{0x119e4, 0x11a01, 29},
@@ -6198,6 +6334,7 @@ var _Other_Alphabetic = &RangeTable{
 		{0x16f4f, 0x16f51, 2},
 		{0x16f52, 0x16f87, 1},
 		{0x16f8f, 0x16f92, 1},
+		{0x16ff0, 0x16ff1, 1},
 		{0x1bc9e, 0x1e000, 9058},
 		{0x1e001, 0x1e006, 1},
 		{0x1e008, 0x1e018, 1},
@@ -6243,8 +6380,9 @@ var _Other_Grapheme_Extend = &RangeTable{
 	R32: []Range32{
 		{0x1133e, 0x11357, 25},
 		{0x114b0, 0x114bd, 13},
-		{0x115af, 0x1d165, 48054},
-		{0x1d16e, 0x1d172, 1},
+		{0x115af, 0x11930, 897},
+		{0x1d165, 0x1d16e, 9},
+		{0x1d16f, 0x1d172, 1},
 		{0xe0020, 0xe007f, 1},
 	},
 }
@@ -6562,6 +6700,7 @@ var _Sentence_Terminal = &RangeTable{
 		{0x115c9, 0x115d7, 1},
 		{0x11641, 0x11642, 1},
 		{0x1173c, 0x1173e, 1},
+		{0x11944, 0x11946, 2},
 		{0x11a42, 0x11a43, 1},
 		{0x11a9b, 0x11a9c, 1},
 		{0x11c41, 0x11c42, 1},
@@ -6679,11 +6818,12 @@ var _Terminal_Punctuation = &RangeTable{
 		{0x11239, 0x1123c, 1},
 		{0x112a9, 0x1144b, 418},
 		{0x1144c, 0x1144d, 1},
-		{0x1145b, 0x115c2, 359},
-		{0x115c3, 0x115c5, 1},
+		{0x1145a, 0x1145b, 1},
+		{0x115c2, 0x115c5, 1},
 		{0x115c9, 0x115d7, 1},
 		{0x11641, 0x11642, 1},
 		{0x1173c, 0x1173e, 1},
+		{0x11944, 0x11946, 2},
 		{0x11a42, 0x11a43, 1},
 		{0x11a9b, 0x11a9c, 1},
 		{0x11aa1, 0x11aa2, 1},
@@ -6703,8 +6843,8 @@ var _Terminal_Punctuation = &RangeTable{
 
 var _Unified_Ideograph = &RangeTable{
 	R16: []Range16{
-		{0x3400, 0x4db5, 1},
-		{0x4e00, 0x9fef, 1},
+		{0x3400, 0x4dbf, 1},
+		{0x4e00, 0x9ffc, 1},
 		{0xfa0e, 0xfa0f, 1},
 		{0xfa11, 0xfa13, 2},
 		{0xfa14, 0xfa1f, 11},
@@ -6713,11 +6853,12 @@ var _Unified_Ideograph = &RangeTable{
 		{0xfa28, 0xfa29, 1},
 	},
 	R32: []Range32{
-		{0x20000, 0x2a6d6, 1},
+		{0x20000, 0x2a6dd, 1},
 		{0x2a700, 0x2b734, 1},
 		{0x2b740, 0x2b81d, 1},
 		{0x2b820, 0x2cea1, 1},
 		{0x2ceb0, 0x2ebe0, 1},
+		{0x30000, 0x3134a, 1},
 	},
 }
 
@@ -7088,6 +7229,8 @@ var _CaseRanges = []CaseRange{
 	{0xA7C4, 0xA7C4, d{0, -48, 0}},
 	{0xA7C5, 0xA7C5, d{0, -42307, 0}},
 	{0xA7C6, 0xA7C6, d{0, -35384, 0}},
+	{0xA7C7, 0xA7CA, d{UpperLower, UpperLower, UpperLower}},
+	{0xA7F5, 0xA7F6, d{UpperLower, UpperLower, UpperLower}},
 	{0xAB53, 0xAB53, d{-928, 0, -928}},
 	{0xAB70, 0xABBF, d{-38864, 0, -38864}},
 	{0xFF21, 0xFF3A, d{0, 32, 0}},
@@ -7711,7 +7854,8 @@ var foldLl = &RangeTable{
 		{0xa7b0, 0xa7b4, 1},
 		{0xa7b6, 0xa7be, 2},
 		{0xa7c2, 0xa7c4, 2},
-		{0xa7c5, 0xa7c6, 1},
+		{0xa7c5, 0xa7c7, 1},
+		{0xa7c9, 0xa7f5, 44},
 		{0xff21, 0xff3a, 1},
 	},
 	R32: []Range32{
@@ -7844,8 +7988,10 @@ var foldLu = &RangeTable{
 		{0xa793, 0xa794, 1},
 		{0xa797, 0xa7a9, 2},
 		{0xa7b5, 0xa7bf, 2},
-		{0xa7c3, 0xab53, 912},
-		{0xab70, 0xabbf, 1},
+		{0xa7c3, 0xa7c8, 5},
+		{0xa7ca, 0xa7f6, 44},
+		{0xab53, 0xab70, 29},
+		{0xab71, 0xabbf, 1},
 		{0xff41, 0xff5a, 1},
 	},
 	R32: []Range32{
@@ -7902,7 +8048,7 @@ var foldInherited = &RangeTable{
 	},
 }
 
-// Range entries: 3483 16-bit, 1730 32-bit, 5213 total.
-// Range bytes: 20898 16-bit, 20760 32-bit, 41658 total.
+// Range entries: 3499 16-bit, 1820 32-bit, 5319 total.
+// Range bytes: 20994 16-bit, 21840 32-bit, 42834 total.
 
 // Fold orbit bytes: 88 pairs, 352 bytes
-- 
2.37.3

